### PR TITLE
feat: added websocket in AI #1347

### PR DESCRIPTION
**Pull request by Astitva877**
## Description
Added Web socket to get response in Chatbot #1347 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

File: src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts [hunk 1]
```diff
@@ -1883,6 +1894,96 @@ class RestExplorerViewModel
...
+      headers: componentData.property.request.headers,
+      method: componentData.property.request.method,
+      queryParams: componentData.property.request.queryParams,
+      url: componentData.property.request.url,
+      auth: componentData.property.request.auth,
+    };
+    const socketValue: Socket =
+      await this.aiAssistentWebSocketService.sendPromptMessage({
+        text: prompt,
+        instructions: `You are an AI Assistant, responsible for answering API related queries. Give the response only in markdown format. Only answer questions related to the provided API data and API Management. Give to the point and concise responses, only give explanations when they are asked for. Always follow best practices for REST API and answer accordingly. Utilize the provided api data ${JSON.stringify(
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts:1918]: lets push this instructions to backend for flexibility, previous implementation isn't that correct

File: src/packages/@app/components/App.svelte [hunk 2]
```diff
@@ -70,6 +71,9 @@
     user.subscribe((value) => {
       isloggedIn = value;
     });
+    // Connect WebSocket
+    const wsService = new AiAssistantWebSocketService();
+    wsService.connectWebSocket();
```
Comment (itsmdasifraza) [inline src/packages/@app/components/App.svelte:77]: we can initiate this code in dashboard file

File: src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts [hunk 3]
```diff
@@ -1883,6 +1894,96 @@ class RestExplorerViewModel
...
+      });
+    let updatePromise = Promise.resolve(); // Initialize a promise chain
+    socketValue.off(`aiResponse_${componentData.tabId}`);
+    socketValue?.on(`aiResponse_${componentData.tabId}`, async (response) => {
+      updatePromise = updatePromise.then(async () => {
+        // Check if the conversation already contains the messageId
+        componentData = this._tab.getValue();
+        const existingMessageIndex =
+          componentData.property.request.ai.conversations.findIndex((conv) => {
+            // console.log("Comparing", conv.messageId, "with", response.messageId); // Debugging line
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts:1932]: logs

File: src/packages/@app/services/ai-assistant.ws.service.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  public connectWebSocket = async () => {
+    const socket = io(`${this.apiUrl}/ai-assistant`, {
+      transports: ["websocket"], // Forces WebSocket transport
+      auth: getAuthHeaders(),
+    });
+    socket.on("connect", () => {
+      console.log("Connected to WebSocket server");
+    });
+    socketStore.set(socket);
+  };
```
Comment (itsmdasifraza) [inline src/packages/@app/services/ai-assistant.ws.service.ts:29]: can we add console.error if socket connection fails, console on success not required

**Review by itsmdasifraza — APPROVED** [review]

File: src/packages/@app/components/App.svelte (hunk 2 above)
Comment (Astitva877) [reply src/packages/@app/components/App.svelte]: done

File: src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts (hunk 3 above)
Comment (Astitva877) [reply src/packages/@app/pages/RestExplorer/RestExplorerPage.ViewModel.ts]: done

**Pull request merged**